playwright/_impl/_helper.py

```diff
@@ -231,3 +232,33 @@ def make_dirs_for_file(path: Union[Path, str]) -> None:
     if not os.path.isabs(path):
         path = Path.cwd() / path
     os.makedirs(os.path.dirname(path), exist_ok=True)
+
+
+async def async_writefile(
+    file: Union[str, Path], mode: str, data: Union[str, bytes]
+) -> None:
+    def inner() -> None:
+        with open(file, mode) as fh:
+            fh.write(data)
+
+    loop = asyncio.get_running_loop()
+    await loop.run_in_executor(None, inner)
+
+
+@overload
+async def async_readfile(file: Union[str, Path], mode: Literal["r"]) -> str:
+    ...
+
+
+@overload
+async def async_readfile(file: Union[str, Path], mode: Literal["rb"]) -> bytes:
+    ...
+
+
+async def async_readfile(file: Union[str, Path], mode: str) -> Union[str, bytes]:
```

Comment:
split this into two methods, `async_readfile` that returns `bytes`, `async_read_text_file` that calls `async_readfile` and decodes.

Reply:
Sams answer as above